### DISCUSSION TITLE
Add custom button in ckeditor plugin grapesjs

### DISCUSSION THREAD

**Discussion by JoanCodinachUoc** [GrapesJS/grapesjs#5401] [Q&A]
I use the pluginOptions of the ckeditor plugin  in grapesjs to include the buttons that i want and works fine. Now i want a button in ckeditor that open my angular modal, like i did with rich text editor.

 this.editor.RichTextEditor.add('variables', { icon: '..', result : 'here i open my pop up}

How can i add a button in ckeditor toolbar that can open my angular pop  up ?

**Comment by artf** [comment]
You have to figure that out from the CKEditor's API

**Reply by JoanCodinachUoc** [reply]
I know that the plugin you can pass the ckeditor as a option and this will have my new toolbar, but how i get this ckeditor object ?
